realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -933,15 +934,13 @@ static void TableQuery_BinaryPredicate(JNIEnv *env, jlong nativeQueryPtr, jlongA
             value2 = BinaryData();
         }
         else {
-            jbyte* bytePtr = env->GetByteArrayElements(value, NULL);
-            if (!bytePtr) {
+            if (!bytes.ptr()) {
```

Comment:
@beeender this checks the result of `env->GetLongArrayElements(javaArray, NULL)`. not `value ` itself.